<?xml version="1.0"?>
<launch>
<arg name="controlModePackage" default="PerceptionModification"/>
<arg name="rearAxleFrameId" default="rear_axle_footprint"/>
<arg name="checkTrajectoryOutdated" default="true"/>
<arg name="lookaheadRatio" value="0.3"/>
<arg name="minLookaheadDistance" value="2.0"/>

<arg name="ControlMode" default="4"/> <!--number from tod_msgs/msg/Status.msg-->
<arg name="ManagerTopic" default="/Vehicle/Manager/status_msg"/> <!-- for AV usage -->
<arg name="TrajectoryTopic" default="safe_trajectory_control_command"/> <!-- for AV usage -->


  <!-- rosrun waypoint_follower pure_pursuit -->
  <node pkg="tod_pure_pursuit" type="tod_pure_pursuit_node" name="PurePursuit" output="screen">
        <remap from="status_msg" to="$(arg ManagerTopic)"/>
        <remap from="trajectory" to="/Vehicle/$(arg controlModePackage)/$(arg TrajectoryTopic)"/>
        <remap from="odometry" to="/Vehicle/VehicleBridge/odometry"/>
        <param name="ControlMode" value="$(arg ControlMode)"/> <!--number from include/tod_msgs/controlMode.h-->
        <param name="rearAxleFrameId" value="$(arg rearAxleFrameId)"/> 
        <param name="lookaheadRatio" value="$(arg lookaheadRatio)"/> 
        <param name="minLookaheadDistance" value="$(arg minLookaheadDistance)"/> 
        <param name="checkTrajectoryOutdated" value="$(arg checkTrajectoryOutdated)"/> 
        <param name="debug" value="false"/>
  </node>
</launch>
